src/services/severa-api-service.ts

```diff
@@ -33,200 +34,113 @@ export const CreateSeveraApiService = (): SeveraApiService => {
   return {
     /**
      * Gets flextime by severaUserId
-     * 
+     *
      * @param severaUserId  Severa user id
      * @returns Flextime of a user
      */
     getFlextimeBySeveraUserId: async (severaUserId: string) => {
-
       const eventDateYesterday = DateTime.now().minus({ days: 1 }).toISODate();
-  
+
       const url = `${baseUrl}/v1/users/${severaUserId}/flextime?eventdate=${eventDateYesterday}`;
 
       const response = await fetch(url, {
         method: "GET",
         headers: {
           Authorization: `Bearer ${await getSeveraAccessToken()}`,
-          "Client_Id": process.env.SEVERA_DEMO_CLIENT_ID,
-          "Content-Type": "application/json",
-        },
+          Client_Id: process.env.SEVERA_DEMO_CLIENT_ID,
+          "Content-Type": "application/json"
+        }
       });
 
       if (!response.ok) {
-        throw new Error(
-          `Failed to fetch flextime: ${response.status} - ${response.statusText}`,
-        );
+        throw new Error(`Failed to fetch flextime: ${response.status} - ${response.statusText}`);
       }
       return response.json();
     },
-      
-    /**
-    * Retrieves the test user information from environment variables.
-    * 
-    * @returns An object containing the test user's email and GUID (ID).
-    * @throws Error if the required environment variables (SEVERA_TEST_USER_ID or SEVERA_TEST_USER_EMAIL) are not set.
-    */
-    getTestUser : async ()  => {
-      if (!process.env.SEVERA_TEST_USER_ID|| !process.env.SEVERA_TEST_USER_EMAIL) {
-        throw new Error("SEVERA_TEST_USER_EMAIL environment variable is missing.");
-      }
-      return {
-      email: process.env.SEVERA_TEST_USER_EMAIL,
-      guid: process.env.SEVERA_TEST_USER_ID 
-      };
-    },
 
     /**
-    * Fetches a Severa user by their email address and checks/updates the 'isSeveraOptIn' status.
-    * 
-    * @param email The email address of the user to be retrieved.
-    * @param attribute Attributes to be checked/updated for the user.
-    * 
-    * @returns The user's Severa GUID, 'isSeveraOptIn' status, and email.
-    */
-    getUserByEmail: async (email: string, attribute: Record<string, string[]>) => {
+     * Fetches a Severa user by their email address and checks/updates the 'isSeveraOptIn' status.
+     *
+     * @param email The email address of the user to be retrieved.
+     * @param keyword Keywords to be checked/updated for the user.
+     *
+     * @returns The user's Severa GUID, 'isSeveraOptIn' status, and email.
+     */
+    getUserByEmail: async (email: string, keyword: Record<string, string[]>) => {
```

Comment:
So you have broken down the keycloak methods into separate functions which is great. This is like the higher level function calling the various methods to basically find user, update user, potentially create keyword if needed etc. So it should be renamed to something else more accurate, potentially- "optInSeveraUser"? 

Also 
I don't think this function needs to be in the service as it's combining various methods of the service and applying logic. I'd suggest creating a src/utils/severa/index.ts and moving this function in there. (If this seems to cause a lot of problems let me know but it should be simple to move).